### PR TITLE
Switch to a more sensible normalization strategy for loss metrics

### PR DESCRIPTION
**Pull request by allevitan**
This makes loss metrics a bit more interpretable. I may put some more work into writing better normalizations if I have a similar internet-free train ride in the future, where I can do some cdtools work but don't have a stable connection to any GPUS...

Key notes:

* This affects all reported loss metrics, so after merging this PR, expect the loss metrics for typical reconstructions to change.

Changes:

* Adds a more flexible "normalizer" class to enable different kinds of normalization for the calculated loss metrics when used for reconstructions.
* Removes the intensity-based normalization for the amplitude MSE and switches to reporting a "straight up" amplitude MSE. I have come to believe this is more interpretable in most cases.
* Switches to use a more reasonable normalization and offset correction for the Poisson NLL results, so that they stop being insane values and never go negative anymore. This will hopefully help encourage adoption of the Poisson NLL loss metric.
* Added initializations for all three loss metrics to all reconstruction models.
* Adjusted the tests to account for the new loss values, and ensured that the different losses all see test coverage
* Update the docs to match, including the tutorial section.
* Update the appropriate example script (near_field_ptycho.py, data taken with a photon counting detector) to use the Poisson NLL error metric.

### CODE REVIEW

File: src/cdtools/tools/losses/losses.py [hunk 1]
```diff
@@ -40,6 +50,8 @@ def amplitude_mse(intensities, sim_intensities, mask=None):
         A tensor of simulated detector intensities
     mask : torch.Tensor
         A mask with ones for pixels to include and zeros for pixels to exclude
+    use_sum : bool
```
Comment (allevitan) [inline src/cdtools/tools/losses/losses.py:53]: This was not implemented

File: src/cdtools/tools/losses/losses.py [hunk 2]
```diff
@@ -155,17 +282,108 @@ def poisson_nll(
...
+    a statistic that is more interpretable for comparing reconstructions. It
+    performs two operations:
+
+    1. **Offset subtraction**: Subtracts the NLL calculated when comparing
+       measured patterns to themselves (i.e., poisson_nll(data, data)). This
+       represents the best-case scenario and makes the loss non-negative.
+
+    2. **Normalization scaling**: Divides by 0.5 times the count of non-zero
+       pixels in the measured patterns. This is because, roughly, each non-zero
+       pixel is expected to contribute to the Poisson NLL, if Poisson noise were
```
Comment (allevitan) [inline src/cdtools/tools/losses/losses.py]: should be "...contribute 0.5 to..."

File: tests/test_reconstructors.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
+from numbers import Number
 import pytest
 import time
 import cdtools
 import torch as t
 import numpy as np
 import pickle
-from matplotlib import pyplot as plt
```
Comment (allevitan) [inline tests/test_reconstructors.py:7]: Why was this deleted? Isn't it needed?

**Review by allevitan — COMMENTED** [review]
Make the fixes, then okay to merge

**Review by allevitan — COMMENTED** [review]
Do a final run of the tests, and then go for merge
